### ISSUE TITLE
Show the query together with the result

### ISSUE THREAD

**Issue by tssm** [nanotee/sqls.nvim#6]
Sometimes I run more than one complex query and after a while navigating their output may become hard if I forget which query corresponds to which output. Is it possible to include it above or after the result? Or even set it into a buffer variable? I'm trying to do it myself, but I'm not that familiar with the LSP API, and reading the code, it seems [the query is extracted and executed by it without the plugin ever getting it](https://github.com/nanotee/sqls.nvim/blob/main/lua/sqls/commands.lua#L25), am I right? If so, I would appreciate any idea of how to proceed.

Thanks beforehand

**Comment by nanotee** [comment]
>it seems [the query is extracted and executed by it without the plugin ever getting it](https://github.com/nanotee/sqls.nvim/blob/main/lua/sqls/commands.lua#L25), am I right?

Yup, I only send the range coordinates to the language server and let it handle the rest

> Is it possible to include it above or after the result? Or even set it into a buffer variable?

They could be inserted in the [table passed to `nvim_buf_set_lines()`](https://github.com/nanotee/sqls.nvim/blob/ade34622050a4a1a50b9db1229a0a60d64e69bdb/lua/sqls/commands.lua#L16) (but really big queries should probably be cut off to avoid making the output too noisy). I think the hardest part is extracting a char-wise range from a buffer because Neovim doesn't have `nvim_buf_get_text()` yet

**Comment by tssm** [comment]
Thanks for the answer! I was thinking about long queries too. I know dadbod creates and alternate file that I was using to retrieve it, but now I'm more incline to print a reference to the query in the result buffer, something like `/path/to/file:lnr:cnr`, which, if I understand correctly, we already have. Does it sound like something you would be interested in getting implemented?

**Comment by nanotee** [comment]
>I know dadbod creates and alternate file that I was using to retrieve it

I had no idea dadbod did this, thanks for the tip!

> Does it sound like something you would be interested in getting implemented?

Definitely! That sounds like a nice UX improvement

**Comment by tssm** [comment]
It is not documented, I found it on a comment on the issue tracker if I remember correctly. You can take the name of a dadbod buffer, change its extension to `.sql` and on that file you will find the query. I used it to show the query in the status line, but as you mentioned, long queries get problematic. I'll give a try to a "jump to query in the previous window" feature
